### PR TITLE
start mission button changed

### LINKED ISSUE #62: remove "start mission" confirmation. just let them start the mission (make it active)
issue:
after watching students use, i think this is annoying.
and it was orginally built to track students active and compare to the mission time and say things like "you have 5 mins left to complete this mission" but that feature hasnt been built yet. so its just an annoying ui atm

possible solution:
- lets comment it out for now, until we add those time tracking features.
- so "start mission" play button should do all the same actions that the confirm "yes, start mission" dialog did.

Image

### PR DESCRIPTION
**Pull request by tairea**
just changed the start mission logic, so it starts the next mission straight away instead of the previous start mission dialog ui.

decided to keep that old button with dialog incase we decide to revert back. (this was for an upcoming feature that timed how long students are taking to complete a mission, and compare it to the teachers estimated mission duration)

fixes #62

### CODE REVIEW

File: src/components/Dialogs/StartMissionDialogV2.vue [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    mdiInformationVariant,
+    mdiCheck,
+    mdiClose,
+    dialog: false,
+    loading: false,
+    currentCourse: null,
+    currentTopic: null,
+  }),
+  computed: {
+    ...mapState(useRootStore, ["currentCourseId", "currentTopicId", "person"]),
```
Comment (charsleysa) [inline src/components/Dialogs/StartMissionDialogV2.vue:44]: What's the difference between the topicId prop and the currentTopicId state? If they're the same then prefer the prop. Also, the courseId should use prop too in that case.

**Review by charsleysa — CHANGES_REQUESTED** [review]

File: src/components/Dialogs/StartMissionDialogV2.vue (hunk 1 above)
Comment (tairea) [reply src/components/Dialogs/StartMissionDialogV2.vue]: removed currentTopicId in favour of propped topicId.
attempted to do the same for currentCourseId but this was breaking as touches many different places.
so left currentCourseId as is.

**Comment by tairea** [comment]
keen to get this one merged into main.
as some the bugs i fixed are bugs in main

File: src/components/Dialogs/CreateEditDeleteNodeDialog.vue [hunk 2]
```diff
@@ -324,6 +324,22 @@ export default {
     dark() {
       return this.$vuetify.theme.isDark;
     },
+    sortTopics() {
```
Comment (charsleysa) [inline src/components/Dialogs/CreateEditDeleteNodeDialog.vue:327]: The word sort is being used as a verb, properties should only use nouns and adjectives. Rename to `sortedTopics`

File: src/components/SolarSystemView/MissionsList/MissionsCard.vue [hunk 3]
```diff
@@ -96,7 +96,8 @@
 
           <div v-if="unlocked" class="d-flex justify-center">
             <!-- Start Mission button -->
-            <StartMissionDialog
+            <StartMissionDialogV2
+              :courseId="courseId"
```
Comment (charsleysa) [inline src/components/SolarSystemView/MissionsList/MissionsCard.vue:100]: Dialog has no courseId prop

**Review by charsleysa — CHANGES_REQUESTED** [review]

**Comment by tairea** [comment]
thanks @charsleysa. u da man

**Pull request merged**